src/logger/LoggerFactory.ts

```diff
@@ -14,50 +14,80 @@
  * limitations under the License.
  */
 
-import { resolve } from 'path';
+import { join, resolve } from 'path';
 import * as winston from 'winston';
 import { FileTransportInstance } from 'winston/lib/winston/transports';
-import Logger from './Logger';
+import { BootstrapUtils } from '../service';
+import { Logger } from './Logger';
 import { LogType } from './LogType';
 
-export default class LoggerFactory {
+export class LoggerFactory {
     private static readonly consoleTransport = new winston.transports.Console({
         format: winston.format.combine(
             winston.format.timestamp(),
             winston.format.cli(),
-            winston.format.printf(LoggerFactory.logFormatTemplate),
+            winston.format.printf((i) => `${i.timestamp} ${i.level} ${i.message}`),
         ),
     });
 
-    private static readonly fileTransport = (id: LogType): FileTransportInstance =>
+    private static readonly consoleOnlyTransport = new winston.transports.Console({
+        format: winston.format.combine(
+            winston.format.printf((i) => {
+                if (i.level.includes('info')) {
+                    return `${i.message}`;
+                } else {
+                    return `${i.message} (${i.level.toUpperCase()})`;
+                }
+            }),
+        ),
+    });
+
+    private static readonly fileTransport = (fileName: string): FileTransportInstance =>
         new winston.transports.File({
-            format: winston.format.combine(winston.format.timestamp(), winston.format.printf(LoggerFactory.logFormatTemplate)),
+            format: winston.format.combine(
+                winston.format.timestamp(),
+                winston.format.printf((i) => `${i.timestamp} ${i.level} ${i.message}`),
+            ),
             options: { flags: 'w' },
-            filename: resolve(LoggerFactory.getLogFileName(id)),
+            filename: resolve(fileName),
             level: 'info',
         });
 
-    public static getLogger(id = LogType.System): Logger {
+    public static getLogger(id: LogType, workingDir = BootstrapUtils.defaultWorkingDir): Logger {
```

Comment:
wouldn't it be more natural to allow user to specify transports like --log Console|File instead of creating bespoke combinations?

Reply:
Linux confuses '|' with the pipe operation. I've used ',' instead. 
Alternately, we could use --logger console --logger file